src/plugins/workspace/public/plugin.ts

```diff
@@ -74,11 +77,36 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
     });
   }
 
-  public async setup(core: CoreSetup, { savedObjectsManagement }: WorkspacePluginSetupDeps) {
+  /**
+   * If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu and disable
+   */
+  private disableManagementApps(core: CoreSetup, management: ManagementSetup) {
+    const currentWorkspaceId$ = core.workspaces.currentWorkspaceId$;
+    this.managementCurrentWorkspaceIdSubscription?.unsubscribe();
+
+    this.managementCurrentWorkspaceIdSubscription = currentWorkspaceId$.subscribe(
+      (currentWorkspaceId) => {
+        if (currentWorkspaceId) {
+          const managementSectionApps = management.sections.section.opensearchDashboards.getAppsEnabled();
+          const disabledApps = managementSectionApps.filter(
+            (app) => app.id === 'settings' || app.id === 'dataSources'
+          );
+          disabledApps?.forEach((app) => app.disable());
+        }
+      }
+    );
+  }
+
+  public async setup(
+    core: CoreSetup,
+    { savedObjectsManagement, management }: WorkspacePluginSetupDeps
+  ) {
     const workspaceClient = new WorkspaceClient(core.http, core.workspaces);
     await workspaceClient.init();
     core.application.registerAppUpdater(this.appUpdater$);
 
+    //  Hide advance settings and dataSource menus and disable in setup
+    this.disableManagementApps(core, management);
```

Comment:
the method name sounds like it will disable whole managementApp

Reply:
Do you have any recommendations? I just want to represent that this will disable some of management apps.

Reply:
i missed the `s` of this method name `disableManagementApps`, it's fine to with this name.